include/jemalloc/internal/tsd_win.h

```diff
@@ -78,6 +96,12 @@ tsd_boot0(void) {
 	}
 	_malloc_tsd_cleanup_register(&tsd_cleanup_wrapper);
 	tsd_wrapper_set(&tsd_boot_wrapper);
+#if JEMALLOC_WIN32_TLSGETVALUE2
+	tgv2_mod = LoadLibraryA("api-ms-win-core-processthreads-l1-1-8.dll");
+	if (tgv2_mod != NULL) {
+		tlsgetvalue2 = (TGV2)GetProcAddress(tgv2_mod, "TlsGetValue2");
```

Comment:
`tlsgetvalue2` is a rather new function, but here it assumes this function always exist when `_M_ARM64EC` is not defined. That means `tlsgetvalue2` could be NULL here. Maybe regress to `TlsGetValue` for that case? Can also do a check in the configuration phase like [this](https://github.com/jemalloc/jemalloc/blob/5b72ac098abce464add567869d082f2097bd59a2/configure.ac#L405).

Reply:
This code doesn't assume existence of TlsGetValue2. Windows APIs `LoadLibrary` and `GetProcAddress` are equivalent to Linux `dlopen` and `dlsym`, so we are basically querying the existence of the API and if so, call it.

Reply:
The reason why we don't want to call TlsGetValue2 if compiled for Arm64EC is that indirect calls in Arm64EC are expensive. Please see microbenchmark results in my PR description for more details. For more information about Arm64EC indirect call checks, please refer to https://learn.microsoft.com/en-us/windows/arm/arm64ec-abi#call-checkers

Reply:
Thanks for clarifying. I overlooked that the check is done in `tsd_wrapper_get`. I still wonder if it is possible to do the detection statically, i.e., during the configuration phase. `tsd_wrapper_get` is on the fastpath where we try to eliminate branching as much as possible. If we can get rid of `if (tlsgetvalue2 != NULL)` there, the perf may be even better.

Reply:
Np. Unfortunately, that runtime check can't be removed because an app built with jemalloc may be installed and run on older Windows OSes that don't support TlsGetValue2.

Reply:
Plus, tlsgetvalue2 != NULL should always evaluate to true if on win11 24h2+ or false otherwise, which means it's great for CPU branch prediction and speculatively execution. It's unlikely to be a perf issue.

Reply:
I see. Agreed.